lib/lz4.h

```diff
@@ -48,6 +48,23 @@ extern "C" {
  * A library is provided to take care of it, see lz4frame.h.
 */
 
+/*-***************************************************************
+*  Export parameters
+*****************************************************************/
+/*!
+*  LZ4_DLL_EXPORT :
+*  Enable exporting of functions when building a Windows DLL
+*/
+#if defined(_WIN32)
```

Comment:
MinGW also support `dllexport`/`dllimport`: http://www.mingw.org/wiki/sampledll

If `dllexport` isn't used, all functions are exported by default. If `dllexport` is used, functions marked as `dllexport` are only exported.

For example, `LZ4_XXH32` is also exported when we don't use `dllexport`.

I think that there is no problem that we use `dllexport` with MinGW.